File: backend/Models/PerfilPontoColeta.cs
namespace AppReciclagem.Models
{
    public class PerfilPontoColeta
    {
        public PerfilPontoColeta(
            int idPontoColeta,
            string nomePontoColeta,
            bool favoritado,
            decimal? avaliacao,
            IEnumerable<MaterialColeta> materiais,
            string descricao)
        {
            IdPontoColeta = idPontoColeta;
            NomePontoColeta = nomePontoColeta;
            Favoritado = favoritado;
            Avaliacao = avaliacao;
            Materiais = materiais;
            Descricao = descricao;
        }

        public int IdPontoColeta { get; set; }

        public string NomePontoColeta { get; set; }

        public bool Favoritado { get; set; }

        public decimal? Avaliacao { get; set; }

        public IEnumerable<MaterialColeta> Materiais { get; set; }

        public string Descricao { get; set; }
    }
}

File: frontend/app/(tabs)/services/models/perfilPontoColeta.ts
export default interface PerfilPontoColeta {
    idPontoColeta: number;
    nomePontoColeta: string;
    favoritado: boolean;
    avaliacao?: number;
    materiais: Material[];
    descricao: string; 
}

interface Material {
    id: number;
    tipoMaterial: string;
    medida: string;
    preco: number;
}